src/operator/contrib/nnvm_to_onnx.cc

```diff
@@ -382,31 +387,16 @@ void ConvertBatchNorm(NodeProto* node_proto, const NodeAttrs& attrs,
   AttributeProto* const spatial = node_proto->add_attribute();
   spatial->set_name("spatial");
   spatial->set_type(AttributeProto::INT);
-  spatial->set_i(1);
-
-  AttributeProto* const consumed = node_proto->add_attribute();
-  consumed->set_name("consumed_inputs");
-  consumed->set_type(AttributeProto::INTS);
-
-  for (int i = 0; i < 5; i++) {
-    int val = (i < 3) ? 0 : 1;
-    consumed->add_ints(static_cast<int64>(val));
-  }
+  // MXNet computes mean and variance per feature for batchnorm.  Enabling spatial mode
+  // (default in ONNX3) implies running batchnorm on all spatial features so we need to explicitly
+  // disable this for MXNet's BatchNorm.
+  spatial->set_i(0);
```

Comment:
Can we add a test for this in case the default behaviour changes in future releases?

Reply:
Good call out, but the test that has been added will cover this.  Batchnorms are in resnet and the math would be way off if the default values didn't align.

One of the reasons I converted to ONNX 3 opset 8 was that it seemed to align well with MXNet's defaults which means we were able to reduce code in more places, but there were still a few customizations like this one required.